parsing/pprintast.ml

```diff
@@ -326,6 +326,9 @@ and core_type1 ctxt f x =
              | _ -> list ~first:"(" ~last:")@;" (core_type ctxt) ~sep:",@;" f l)
           l longident_loc li
     | Ptyp_variant (l, closed, low) ->
+        let first_is_inherit = match l with
+            {Parsetree.prf_desc = Rinherit _}::_ -> true
```

Comment:
Most of the time, relatively recent code in the compiler use `|` even on the first clause of the pattern matching.